libhwmon/wrapadl.cpp

```diff
@@ -213,7 +280,8 @@ int wrap_adl_get_power_usage(wrap_adl_handle *adlh, int gpuindex, unsigned int*
 		return -1;
 	}
 	int power = 0;
-	rc = adlh->adl2Overdrive6CurrentPowerGet(adlh->context, adlh->phys_logi_device_id[gpuindex], 0, &power);
+	int adlidx = adlh->opencl_adl_device_id[gpuindex];
```

Comment:
gpuindex should be always adl id, the caller should use the mapping array if needed